src/Compilers/Core/CodeAnalysisTest/Text/TextChangeTests.cs

```diff
@@ -220,5 +221,97 @@ public void TestGetTextChangesToChangedText()
                 Assert.Equal(expected.NewText, actual.NewText);
             }
         }
+
+        sealed class TextLineEqualityComparer : IEqualityComparer<TextLine>
+        {
+            public bool Equals(TextLine x, TextLine y)
+            {
+                return x.Span == y.Span;
+            }
+
+            public int GetHashCode(TextLine obj)
+            {
+                return obj.Span.GetHashCode();
+            }
+        }
+
+        [Fact]
+        public void TestOptimizedSourceTextLinesSimpleSubstition()
+        {
+            var changedText = SourceText.From("Line1\r\nLine2\r\nLine3").WithChanges(
+                new TextChange(new TextSpan(8, 2), "IN"),
+                new TextChange(new TextSpan(15, 2), "IN"));
+            Assert.Equal(SourceText.From(changedText.ToString()).Lines, changedText.Lines, new TextLineEqualityComparer());
```

Comment:
You could create a helper method to avoid duplication in the various test cases:

    private static void AssertChangedTextLinesHelper(string originalText, params TextChange[] changes)
    {
        var changedText = SourceText.From(originalText).WithChanges(changes);
        Assert.Equal(SourceText.From(changedText.ToString()).Lines, changedText.Lines, new TextLineEqualityComparer());
    }